### ISSUE TITLE
Dataset processing error-ing on `file_info = file_url.file_info()`

### ISSUE THREAD

**Issue by maniarathi** [chanzuckerberg/single-cell-data-portal#1928]
Three datasets recently failed to upload all with the same error:
https://czi-sci.slack.com/archives/C0244PQK934/p1644017956507849
https://czi-sci.slack.com/archives/C0244PQK934/p1644017976627249
https://czi-sci.slack.com/archives/C0244PQK934/p1644017977605369

The stack trace for these are the same:

```
Traceback (most recent call last):
File "/backend/corpora/dataset_processing/process.py", line 554, in main
process(dataset_id, os.environ["DROPBOX_URL"], os.environ["ARTIFACT_BUCKET"])
File "/backend/corpora/dataset_processing/process_download_validate.py", line 37, in process
local_filename = download_from_dropbox_url(
File "/backend/corpora/dataset_processing/process.py", line 265, in download_from_dropbox_url
file_info = file_url.file_info()
File "/backend/corpora/common/utils/dl_sources/url.py", line 83, in file_info
resp.raise_for_status()
File "/usr/local/lib/python3.8/dist-packages/requests/models.py", line 960, in raise_for_status
raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url:
```

It seems like these error may be transient (need to verify with curators), but we should look into ensuring that either the logging for this error is better or that we have a failsafe if the URL returns a 500 for the `requests.head` call. Alternatively, I believe we can remove this code per the recommendation in [this issue](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/1910).

**Comment by ebezzi** [comment]
Turns out the `file_info` call is actually used in the upload pipeline for a few things, in particular by the progress indicator, so we might want to keep it after all. I don't know what caused the 500 here, but it could have been an excess of calls to Dropbox (since we uploaded a lot of datasets that day? I am not sure).

To mitigate, we could add retries to that call, or make it optional (although that would require disabling the progress indicator)
